beater/pub_test.go

```diff
@@ -11,9 +11,11 @@ import (
 )
 
 func TestNumberGoRoutines(t *testing.T) {
+	runtime.GC() // force goroutine accounting?
 	before := runtime.NumGoroutine()
 	_, err := newPublisher(pip{}, 1, time.Duration(0))
 	assert.NoError(t, err)
+	runtime.GC()
```

Comment:
What do you think about something like `wait_until_true_with_a_timeout`, eg. 
```
func TestNumberGoRoutines(t *testing.T) {
	expected := runtime.NumGoroutine() + runtime.GOMAXPROCS(0)
	_, err := newPublisher(pip{}, 1, time.Duration(0))
	assert.NoError(t, err)
	for i := 0; i < 10; i++ {
		if expected == runtime.NumGoroutine() {
			break
		} else {
			time.Sleep(100 * time.Millisecond)
		}
	}
	assert.Equal(t, expected, runtime.NumGoroutine())
}
```

It is also not very nice, but I'd prefer it over adding `sync.Wait` to production code for making a test work.

Reply:
That's actually where I started but was considered this will not catch the case where `NumGoroutine` continues to grow after the condition is already met.